LFS-1098: After a vocabulary is installed, all its nodes should be checked in

IntegralProgrammer opened the pull request:
This PR implements LFS-1098 by keeping an internal list of all JCR nodes added from the installation of a vocabulary (either from BioPortal or from a local file). After committing the nodes to JCR, the internal list is used to check-in each node so that they have a version history.

To test:
1. Build this branch
2. Start CARDS in the `dev` runmode with a valid BioPortal API key
3. Install a vocabulary from BioPortal and inspect it in Composum (under `/Vocabularies/`). Ensure that both the root node of the vocabulary (eg. `HANCESTRO`) and all descendant nodes (eg. `Aruba`, `Australia`, `Austria`, etc...) have the property `jcr:isCheckedOut` set to `false`.
4. Install a vocabulary from a local file, inspect it in Composum and ensure that the above test passes.

IntegralProgrammer committed:
LFS-1098: After a vocabulary is installed, all its nodes should be checked in

Check in all vocabulary nodes for a vocabulary installed from BioPortal

IntegralProgrammer committed:
LFS-1098: After a vocabulary is installed, all its nodes should be checked in

Check in all vocabulary nodes for a vocabulary installed from a local file

sdumitriu reviewed (changes requested):
I think it would be better to put all the needed changes in `OntologyIndexerUtils`:

- move the thread local member in there, and instead of explicitly initializing it, create it with `private ThreadLocal<List<Node>> nodesToCheckIn = ThreadLocal.withInitial(ArrayList::new);`
- don't change the signature of `createVocabularyTermNode`, since the list is already in the utils object
- do the checkin in the `saveSession` method
- also clear thread local after checking in with `this.nodesToChecking.remove()` to free up memory
- no changes required to the two indexer classes
- maybe rename the `saveSession` method, which is too specific as it is, to `commit`
WDYT?

IntegralProgrammer committed:
LFS-1098: After a vocabulary is installed, all its nodes should be checked in

Move all vocabulary check-in code to the OntologyIndexerUtils.java file

IntegralProgrammer committed:
LFS-1098: After a vocabulary is installed, all its nodes should be checked in

Create the OntologyIndexerUtils.finalizeInstall method which, after creating all nodes for the installation of a vocabulary, saves the JCR session and checks in all these new nodes.

IntegralProgrammer commented:
> I think it would be better to put all the needed changes in `OntologyIndexerUtils`:
> 
>     * move the thread local member in there, and instead of explicitly initializing it, create it with `private ThreadLocal<List<Node>> nodesToCheckIn = ThreadLocal.withInitial(ArrayList::new);`
> 
>     * don't change the signature of `createVocabularyTermNode`, since the list is already in the utils object
> 
>     * do the checkin in the `saveSession` method
> 
>     * also clear thread local after checking in with `this.nodesToChecking.remove()` to free up memory
> 
>     * no changes required to the two indexer classes
> 
>     * maybe rename the `saveSession` method, which is too specific as it is, to `commit`
>       WDYT?

- Moved all needed changes to `OntologyIndexerUtils`
- Created a `OntologyIndexerUtils.finalizeInstall` method which saves the JCR session and then checks in all the newly installed vocabulary nodes.
- `BioOntologyIndexer.java` and `FileUploadIndexer.java` remain unchanged except for replacing `saveSession` with `finalizeInstall` and updated Javadoc.
